server/util/clickhouse/clickhouse.go

```diff
@@ -182,19 +184,30 @@ func (h *DBHandle) FlushInvocationStats(ctx context.Context, ti *tables.Invocati
 	var lastError error
 	for retrier.Next() {
 		res := h.DB(ctx).Create(inv)
+		lastError = res.Error
 		if res.Error == nil {
-			return nil
+			break
 		}
 		if errors.Is(res.Error, syscall.ECONNRESET) || errors.Is(res.Error, syscall.ECONNREFUSED) {
 			// Retry since it's an transient error.
-			lastError = res.Error
 			log.Infof("attempt to write invocation (id: %q) to clickhouse failed: %s", res.Error, ti.InvocationID)
 			continue
 		} else {
-			return res.Error
+			break
 		}
```

Comment:
I think you can simplify this like:

```
		if errors.Is(res.Error, syscall.ECONNRESET) || errors.Is(res.Error, syscall.ECONNREFUSED) {
			// Retry since it's an transient error.
			log.Infof("attempt to write invocation (id: %q) to clickhouse failed: %s", res.Error, ti.InvocationID)
			continue
		}
		break
}

```

Reply:
Done